### PR TITLE
tekton task to run unit tests

### LINKED ISSUE #532: Create `Task` for Tekton Pipeline unit tests
## Expected Behavior
We should be testing and releasing Pipelines with Pipelines.

We should have a `Task` (and related Resources) defined that we can use to
- Trigger [all unit tests in build-pipelines](https://github.com/knative/build-pipeline/tree/master/test#unit-tests)
- Write results to [gubernator](https://github.com/knative/test-infra/tree/master/ci/gubernator)

### The Task

* Inputs:
  - git repo
* Outputs:
  - some kind of result info which can be passed to #535
* Parameters:
-  PR to update the checks of

(Can decide to make this a generalized "run go tests" task, or can make it Tekton repo specific)

## Actual Behavior

We currently use [prow](https://github.com/knative/test-infra/tree/master/ci/prow) and [presubmit tests](https://github.com/knative/build-pipeline/blob/master/test/presubmit-tests.sh) (i think?) to run unit tests.

## Additional Info

* Actually triggering this is not in scope for this issue, it's fine to test manually (we'll be linking these together in a later issue)

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This adds a Task to run pipeline's unit tests

- This will move to `tektoncd/catalog` later on as it's meant to be a reusable task :angel: 
- This will be hooked-up to the CI once we work on #922

Improvements possible:

- [ ] Save the cover file somewhere (in a resource or *as a resource* :smiling_imp:)
- [x] More parameters (to be able to filter tests, or add build tags)

Closes #532 

cc @bobcatfish 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ :no_good_man: ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ :no_good_man: ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/ci-unit-test.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      type: git
+      targetPath: src/${inputs.params.package}
+  #outputs: # coverage file
+  steps:
+  - name: unit-test
+    image: golang:1.12
+    workingdir: /workspace/src/${inputs.params.package}
+    command:
+    - go 
+    args: ["test", "-v", "-cover", "./..."]
```
Comment (bobcatfish) [inline tekton/ci-unit-test.yaml]: do we want that race condition flag too?

Comment (bobcatfish) [reply tekton/ci-unit-test.yaml]: and/or a flexible set of flags, maybe with a default?

(need that array type! :D https://github.com/tektoncd/pipeline/issues/207)

Comment (vdemeester) [reply tekton/ci-unit-test.yaml]: good point for `race` !
yeah we really need #207 :sweat_smile: as of now, we need a wrapper or some bashism to support flexible set of flags (which reduce a bit readability)

Comment (vdemeester) [reply tekton/ci-unit-test.yaml]: Updated (in a separate commit)

**Comment by vdemeester** [comment]
@bobcatfish removed the `wip` as I feel it could get in like that :angel: 
I'll prepare a PR on [`tektoncd/catalog`](https://github.com/tektoncd/catalog) for some `go` related re-usable tasks (inspired by this one)

**Pull request merged**